components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java

```diff
@@ -1328,4 +1328,22 @@ void addFileToProductDocumentation(APIProductIdentifier prodcutId, Documentation
      * @return List of removed resources that are reused among API Products
      */
     List<APIResource> getRemovedProductResources(Set<URITemplate> updatedUriTemplates, API existingAPI);
+
+    /**
+     * Adds a new Endpoint Registry
```

Comment:
I think it's incorrect to user APIProvider interface for this. We should have a separate interface.  EndpointRegistry

Reply:
and then implement the methods in EndpointRegistryImpl

Reply:
Ack. will do the changes. Used APIProvider as this operations are related to API Provider. Anyway it's better to have separate interface for EndpointRegistry.

Reply:
Fixed in 6de38c7